src/pages/issues.tsx

```diff
@@ -9,36 +8,88 @@ import {
     NO_ISSUES_FOUND_MESSAGE,
 } from '@/components/constants/messages';
 import { ISSUES_URL } from '@/components/constants/url';
+import axios from 'axios';
+
+type SearchFieldProps = {
+    searchText: string;
+    onSearchTextChanged: (event: any) => void;
+    onSearchTextSubmitted: () => void;
+    loading: boolean;
+};
+
+const SearchField = ({
+    searchText,
+    onSearchTextChanged,
+    onSearchTextSubmitted,
+    loading,
+}: SearchFieldProps) => {
+    return (
+        <div className={classNames.searchFieldContainer}>
+            <input
+                placeholder="Enter query string to search issues"
+                value={searchText}
+                onChange={onSearchTextChanged}
+                className={classNames.issueSearchInput}
+            />
+            <button
+                onClick={onSearchTextSubmitted}
+                className={classNames.issuesSearchSubmitButton}
+                disabled={loading || !searchText.trim()}
+            >
+                Submit
+            </button>
+        </div>
+    );
+};
 
 const Issues: FC = () => {
     const [issueList, setIssueList] = useState<[]>([]);
+    const [searchText, setSearchText] = useState('');
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<null | any>(null);
 
-    const { response, error, isLoading } = useFetch(ISSUES_URL, {
-        data: {
-            userData: {
-                roles: {
-                    super_user: true,
-                },
-            },
-        },
-    });
+    const fetchIssues = async () => {
+        try {
+            setIsLoading(true);
+            const res = await fetch(`${ISSUES_URL}?q=${searchText}`);
+            const data = await res.json();
+            if ('issues' in data) {
+                const issuesAndPullRequests: any = data.issues;
+                const onlyIssues = issuesAndPullRequests.filter(
+                    (issue: { hasOwnProperty: (arg0: string) => any }) =>
```

Comment:
Please make this readable

Reply:
Yes, have changed it. Idk why I added `{ hasOwnProperty: (arg0: string) => any }`, it wasn't required.

Reply:
@vvaibhavdesai has this been resolved or do I need to address this in the future enhancement?